### ISSUE TITLE
Accumulation of dependencies

### ISSUE THREAD

**Issue by julifan** [spcl/perf-taint#29]
Right now, dependencies are accumulated in the JSON processing. As in the example below, if a function f (depending on x1) calls a function g (depending on x2), the processed JSON states that f has a dependency on x2.  For using the dependencies and modeling them on a function level, I would need them to not be accumulated (except if a function is called from within a loop that depends on a tainted variable). 

Example: 
```
int x1 EXTRAP;
int x2 EXTRAP;
	
void g()
{
	int add = 0;
	for (int i = 0; i < x2; i++) {
		add += 1;
	}
}

void f(int x1) 
{
	int add = 0;
	for (int i = 0; i < x1; i++) {
		//just du something
		add += 1;
	}
	g();
}

int main(int argc, char ** argv)
{
	x1 = atoi(argv[1]);
	x2 = atoi(argv[2]);
	
	register_variable(&x1, sizeof(x1), VARIABLE_NAME(x1));
	register_variable(&x2, sizeof(x2), VARIABLE_NAME(x2));
	
	f(x1);
}
```

Processed JSON:

```
{
  "functions": {
    "f": {
      "file": "dependency-accumulation.c",
      "func_idx": 0,
      "line": 17,
      "loops": [
        {
          "callstack": [
            [
              2
            ]
          ],
          "data": [
            {
              "dependency": "additive",
              "operands": [
                {
                  "dependency": "unknown",
                  "operands": [
                    "x1"
                  ]
                },
                {
                  "dependency": "additive",
                  "operands": [
                    {
                      "dependency": "additive",
                      "operands": [
                        {
                          "dependency": "unknown",
                          "operands": [
                            "x2"
                          ]
                        }
                      ]
                    }
                  ]
                }
              ]
            }
          ],
          "deps": [
            [
              "x1"
            ],
            [
              "x2"
            ]
          ],
          "not_found_params": null
        }
      ]
    },
    "g": {
      "file": "dependency-accumulation.c",
      "func_idx": 1,
      "line": 9,
      "loops": [
        {
          "callstack": [
            [
              2,
              0
            ]
          ],
          "data": [
            {
              "dependency": "additive",
              "operands": [
                {
                  "dependency": "unknown",
                  "operands": [
                    "x2"
                  ]
                }
              ]
            }
          ],
          "deps": [
            [
              "x2"
            ]
          ],
          "not_found_params": [
            "x1"
          ]
        }
      ]
    }
  },
  "functions_demangled_names": [
    "float",
    "__float128",
    "main",
    "register_variable"
  ],
  "functions_mangled_names": [
    "f",
    "g",
    "main",
    "register_variable"
  ],
  "functions_names": [
    "f",
    "g",
    "main",
    "register_variable"
  ],
  "parameters": [
    "x1",
    "x2"
  ],
  "unused_parameters": [
    "p"
  ]
}```

**Comment by mcopik** [comment]
@julifan I pushed a fix to this in 58de8bc - JSONConverter accepts now a new flag `--no-accumulate-nonloop-dependencies` that allows skipping processing function dependencies outside of a loop. Obviously, this affects the final result of detected parameters in a function.

For details, see the difference between these two test results:
Accumulated: https://github.com/spcl/perf-taint/blob/master/tests/unit/dataflow/function_call_nested.cpp.processed.json
Non-accumulated: https://github.com/spcl/perf-taint/blob/master/tests/unit/dataflow/function_call_nested.cpp.processed2.json

**Comment by mcopik** [comment]
@julifan One more thing - I had to make breaking API changes. From now on, all internal functions have the prefix `perf_taint` (C) or are in the `perf_taint` namespace (C++). So:

```
perf_taint_register_variable
perf_taint::register_variable
```

Furthermore, the internal functions (`register_variable`, `has_label`, `delete_label`) won't show up in JSON results anymore. I hope that's not too much work for you. I was restructuring testing for this change, so it was easier to make both changes at the same time.
